### ISSUE TITLE
values.yaml can't be reused with stratos 

### ISSUE THREAD

**Issue by viovanov** [cloudfoundry-incubator/kubecf#1174]
**Is your feature request related to a problem? Please describe.**
We can't share a yaml with stratos because the `kube.storage_class` key has a different type in each chart.

**Describe the solution you'd like**
Rename the keys or share the type, so the values.yaml can be reused.

**Comment by viovanov** [comment]
ping @colstrom

**Comment by jandubois** [comment]
> We can't share a yaml with stratos

I don't consider sharing a `config-values.yaml` between different charts a good practice. If the 2 charts are so tightly coupled, why aren't they combined into a single chart? Also, is this just about kubecf and stratos, or is this supposed to cover other charts as well?

With helm 3 we have the ability to validate user input using JSON schema files. I think this is a great feature to improve the user experience. When there are any failures in the templating code, helm just barfs up a disgusting mix of stack trace, stringified YAML blobs, and maybe some cryptic hints about the error somewhere in the middle, leaving the user to figure out what helm had for breakfast. Validating input upfront provides much better (but still far from perfect) feedback about what is wrong.

Throwing random data from other charts into the helm configuration makes it impossible to reject unexpected properties, and therefore also impossible to catch typos from user input.

As a compromise, I could (maybe) see that we define a set of commonly shared configuration data, like storage classes, RBAC settings, registry location,, image pull secrets etc, and create a schema for that. Then each chart can be deployed with 2 config files, one of which could be shared between charts:

```
helm install kubecf --values shared-config.yaml --values kubecf-config.yaml
```

Each chart would then include the schema for the shared configuration in its own validation schema. But again, this needs to be carefully planned, as the shared schema will either be non-extensible, or will have to allow for additional properties, making it impossible to catch typos in the shared config parts. It quickly turns into a mess, where the user will wonder which property can go into the shared file, and which one may not (yet?) be supported by all the charts.

So after writing all of this, I still think sharing config files between charts is a bad idea, even in the compromise form. We should still try to develop a shared schema for a subset of shared properties, but the user would have to maintain them in separate files, so they still get the full benefit of schema validation.

I know the SUSE CAP documentation uses a shared config file in its examples, and I think that documentation should be changed because that practice will cease to work in the future, even if we define the properties in the same manner.

**Comment by fargozhu** [comment]
@viovanov @troytop your comments and inputs to @jandubois concerns are highly appreciated.

I would suggest to create another issue  to cover the "...validate user input using JSON schema files...".

**Comment by troytop** [comment]
Preaching to the choir. We've all agreed several times to NOT share values files between helm charts. SUSE CAP documentation should now describe using separate values.yaml files for all components, and I don't think there are any instructions anywhere that tell people to do this with KubeCF and Stratos. 

+1 on JSON schema support for the kubecf chart.

**Comment by jandubois** [comment]
> I would suggest to create another issue to cover the "...validate user input using JSON schema files...".

That was already implemented as part of #1086. I don't think there is anything to do for this issue, so I'm going to close it.
